packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -0,0 +1,295 @@
+
+Pluralization is handled by providing the pluarlization keys required by the language as leaf KVPs, creating a "pluralization context".
+
+**Example English file with pluralization keys:**
+```json5
+{
+  "cars": {  // All the children of `cars` are in a "pluralization context"
+    "one": "I have {count} car",
+    "other": "I have {count} cars"
+  }
+}
+```
+
+Different languages require different sets of pluralization keys. For example, the Polish (`pl`) has 4 required pluralization keys: `one`, `few`, `many`, and `other`. The Polish version of the previous example would be:
+
+**Example Polish file with pluralization keys:**
+```json5
+{
+  "cars": {
+    "one": "Mam {count} samochód",
+    "few": "Mam {count} samochody",
+    "many": "Mam {count} samochodów",
+    "other": "Mam {count} samochodu",
+  }
+}
+```
+
+## Reserved pluralization keys
+
+In order to avoid collisions, the following keys are reserved for use as pluralization keys, and must not be used as keys of leaf KVPs outside of the pluralization context:
+
+* `few`
+* `many`
+* `one`
+* `other`
+* `two`
+* `zero`
+
+These keys are derived from the names given to the pluralization rules of all languages, as defined in the [Unicode Consortium's Common Locale Data Repository (CLDR)](https://github.com/unicode-org/cldr).
+
+This list is current as of verison `37` of the CLDR. In the unlikely event that new pluralization rules names are added, they will be added to this reserved list in a future version of this specification.
+
+**Example INVALID English file:**
+```json5
+{
+  "foo": "bar",
+  "other": "Other" // INVALID: `other` is a keyword reserved for use in pluralization contexts!
+}
+```
+
+## Invalid pluralization keys
+
+Further, pluralization keys that are not required by the language must not be used, even within a pluralization context.
+
+For example, even though `zero` is a required pluralization key for some languages (eg. Arabic), it is not a required pluralization key in English. Therefore, the `zero` key should not be present in an English file alongside the pluralization KVPs required by English.
+
+**Example INVALID English file:**
+```json5
+{
+  "cars": {
+    "zero": "I have no cars", // INVALID: `zero` is not a required pluralization key for English!
+    "one": "I have {count} car",
+    "other": "I have {count} cars"
+  }
+}
+```
+
+## Detecting pluralization contexts
+
+These constraints allow for an algorithm for identifying pluralization contexts:
+
+If a node has child leaf KVPs for each of the pluralization keys required by the language, then its children are within a pluralization context.
+
+Any additional children within the context are invalid.
+
+# Serialization
+
+React I18n can be serialized to different formats. However, not all serialization formats are created equal.
+
+Two serializations are officialy defined. Consumers of React I18n `1.0` files are expected to be able to parse files serialized into each of these serialization formats.
```

Comment:
Ambiguous version here.

Reply:
Changed to `Consumers of React I18n `1.0` schema files` (added `schema`).